src/Forms/TreeDropdownField.php

```diff
@@ -555,6 +570,27 @@ public function getSourceObject()
         return $this->sourceObject;
     }
 
+    protected function flattenChildrenArray($children, $parentTitles = [])
+    {
+        $output = [];
+    
+        foreach ($children as $child) {
+            $childTitles = array_merge($parentTitles, [$child['title']]);
+            $grandChildren = $child['children'];
+            $contextString = implode('/', $parentTitles);
+            
+            $child['contextString'] = ($contextString !== '') ? $contextString .'/' : '';
+            $child['children'] = [];
+    
+            if (in_array($child['id'], $this->realSearchIds)) {
```

Comment:
Probably should also include this child if `!$this->search || in_array($child['id'], $this->realSearchIds)`, otherwise output will be empty.